docs/source/algorithms/SaveP2D-v1.rst

```diff
@@ -0,0 +1,134 @@
+theta, lambda, dSpacing and dSpacingPerpendicular. 
+
+Output
+######
+
+The output file contains a short comment header giving the title, the instrument parameter file, 
+the binning parameters of the workspace and the used instrument/detector bank.
+Thereafter the measuring data is written into 5 columns, namely, 2theta, lambda, dSpacing,
+dSpacingPerpendicular and intensity.
+
+Usage
+-----
+
+**Example: Create a ".p2d" file from a 2D Workspace. Remember to change the Filepath for the OutputFile!**
+
+.. testcode:: SaveP2D
+
+	# create a 2D Workspace
+    # repeat this block for each spectrum
+    xData = [1.0,2.0,3.0,4.0,5.0,6.0,7.0,8.0,9.0]		     # d values for one spectrum (one dPerpendicular value)
+    yData = ['1','2','3','4']					             # dPerpendicular binedges
+    zData = [1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0]		     # intensity values
+    eData = [1,1,1,1,1,1,1,1,1]                              # error values
+
+    # used to join all spectra
+    xDataTotal = []					                         # d Values for all spectra
+    zDataTotal = []					                         # intensity values for all spectra
+    eDataTotal = []                                          # error values for all spectra
+    nSpec = len(yData)-1                                     # number of spectra
+
+    # Create d and intensity lists for workspace
+    for i in range(0,nSpec):
+        xDataTotal.extend(xData)	   # extends the list of x values in accordance to the number of spectra used
+        zDataTotal.extend(zData)	   # extends the list of intensity values in accordance to the number of spectra used
+        eDataTotal.extend(eData)       # extends the list of error values in accordance to the number of spectra used
+
+    # Create a 2D Workspace containing d and dPerpendicular values with intensities
+    CreateWorkspace(OutputWorkspace = 'Usage_Example', DataX = xDataTotal, DataY = zDataTotal, DataE = eDataTotal, WorkspaceTitle = 'test', NSpec = nSpec, UnitX = 'dSpacing', VerticalAxisUnit = 'dSpacingPerpendicular', VerticalAxisValues = yData)
+
+    # Save to the users home directory
+    file_name = "Usage_Example"
+    path = os.path.join(os.path.expanduser("~"), file_name)
+
+    # Create a .p2d file containing the testdata	
+    SaveP2D(Workspace = "Usage_Example", OutputFile = path, RemoveNaN = False, RemoveNegatives = False, CutData = False)
+
+    # Does the file exist? If it exists, print it!
+    path = os.path.join(os.path.expanduser("~"), file_name + '.p2d')
+    if os.path.isfile(path):
+        with open(path, 'r') as of:
+            data = of.readlines()
+        for entry in data:
+            print(entry[:-1]) # leave out the last character to remove unnecessary newlines
+
+.. testcleanup:: SaveP2D
+
+    os.remove(path)
+
+Output:
+The resulting output file (Usage_Example.p2d) looks like this(2theta and lambda get calculated in the algorithm):
+
+.. testoutput:: SaveP2D
+   :options: +ELLIPSIS
```

Comment:
I think this will posssibly correct the whitespace tab issues, but I'm not sure. Worth a try though
```suggestion
   :options: +ELLIPSIS,+NORMALIZE_WHITESPACE
```

Reply:
I tried adding some more whitespaces to the expected output, if that doesnt work I will try your suggestion :)